ui/plugin-system/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx

```diff
@@ -107,9 +107,14 @@ function ListVariableEditorForm({ action, control }: KindVariableEditorFormProps
    * spec that will be used for preview. The reason why we do this is to avoid
    * having to re-fetch the values when the user is still editing the spec.
    */
-  const [previewSpec, setPreviewSpec] = useState<ListVariableDefinition>(form.getValues() as ListVariableDefinition);
+  const [previewSpec, setPreviewSpec] = useState<ListVariableDefinition & { refreshToken: string }>({
+    ...form.getValues(),
+    refreshToken: String(Date.now()),
```

Comment:
I'd try to avoid mixing specs with other values that are specific of the fetching logic.